### PR TITLE
feat: update trips empty state view

### PR DESCRIPTION
**Pull request by daledah**
### Fixed Issues

$ https://github.com/Expensify/App/issues/47937
PROPOSAL: https://github.com/Expensify/App/issues/47937#issuecomment-2308369284

### Tests

Precondition: There are no trip expenses.

1. Open search page
2. Navigate to trips
3. Verify that: The screen displays the updated modal

- [x] Verify that no errors appear in the JS console

### QA Steps

Precondition: There are no trip expenses.

1. Open search page
2. Navigate to trips
3. Verify that: The screen displays the updated modal

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/0a2cffcc-8100-48af-9424-245c55078aab

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/04a3e622-8eff-4807-b4f7-0e4148786fe3

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/c6df03dc-5eaa-4504-9914-6a2322382779

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/dc8ffcf9-4dbe-448e-b0de-3f406b197e6e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/431ab18d-ffc3-4329-9f98-08ce9a8f53d3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/8b07ac04-c52c-4239-8d0d-44e712cd846c

</details>

### CODE REVIEW

**Comment by luacmartins** [comment]
@daledah what's the ETA to get this PR in review?

**Comment by luacmartins** [comment]
@daledah it seems like we just need to make sure the modal is not cut off and we need to update some styles to do that. Do you have any specific questions about that?

**Comment by dubielzyk-expensify** [comment]
This card shouldn't be cut off. We should be showing the full card here and if it can't fit, then the whole view should be able to be scrolled
CleanShot 2024-09-09 at 20 22 57@2x

**Comment by shawnborton** [comment]
Yup, totally agree with that. The inner modal itself should not be scrollable, just the view that wraps it.

**Comment by DylanDylann** [comment]
Give a update in the issue: https://github.com/Expensify/App/issues/47937#issuecomment-2343010997

**Comment by luacmartins** [comment]
@daledah what's the latest here?

**Comment by luacmartins** [comment]
@daledah react compiler is failing

**Comment by shawnborton** [comment]
Looks like the empty state isn't filling up all of the available vertical space when the viewport height is taller:
CleanShot 2024-09-16 at 13 24 14@2x

See the difference here with other empty states:
CleanShot 2024-09-16 at 13 24 53@2x

File: assets/animations/BankVault.lottie
Comment (DylanDylann) [inline assets/animations/BankVault.lottie:1]: What is the purpose of adding this file?

File: assets/animations/GenericEmptyState.lottie
Comment (DylanDylann) [inline assets/animations/GenericEmptyState.lottie:1]: What is the purpose of adding this file?

File: src/pages/Search/EmptySearchView.tsx [hunk 1]
```diff
@@ -1,43 +1,114 @@
...
-                    headerContentStyles: StyleUtils.getWidthAndHeightStyle(variables.w191, variables.h172),
-                    title: translate('search.searchResults.emptyTripResults.title'),
-                    subtitle: translate('search.searchResults.emptyTripResults.subtitle'),
+                    headerContentStyles: StyleUtils.getWidthAndHeightStyle(335, 220),
+                    title: translate('travel.title'),
+                    titleStyles: {...styles.textAlignLeft},
+                    subtitle: subtitleComponent,
                     buttonText: translate('search.searchResults.emptyTripResults.buttonText'),
-                    buttonAction: () => Navigation.navigate(ROUTES.TRAVEL_MY_TRIPS),
+                    buttonAction: () => Navigation.navigate(ROUTES.WORKSPACE_PROFILE_ADDRESS.getRoute(activePolicyID ?? '-1')),
```
Comment (DylanDylann) [inline src/pages/Search/EmptySearchView.tsx:102]: Why do we navigate to WORKSPACE_PROFILE_ADDRESS here? Or I missed this some where?

File: src/pages/Search/EmptySearchView.tsx [hunk 2]
```diff
@@ -1,43 +1,114 @@
...
+                        onPress={() => {
+                            Linking.openURL(CONST.BOOK_TRAVEL_DEMO_URL);
+                        }}
+                    >
+                        {translate('travel.bookADemo')}
+                    </TextLink>
+                    {translate('travel.toLearnMore')}
+                </Text>
+                <View style={[styles.flex1, styles.flexRow, styles.flexWrap, styles.rowGap4, styles.pt4, styles.pl1]}>
+                    {tripsFeatures.map((tripsFeature) => (
```
Comment (DylanDylann) [inline src/pages/Search/EmptySearchView.tsx:67]: Should we use FeatureList Component here?

File: src/pages/Search/EmptySearchView.tsx [hunk 3]
```diff
@@ -1,43 +1,114 @@
...
+    const styles = useThemeStyles();
+
+    const [activePolicyID] = useOnyx(ONYXKEYS.NVP_ACTIVE_POLICY_ID);
+
+    const subtitleComponent = useMemo(() => {
+        const tripsFeatures: TripsFeaturesProps[] = [
+            {
+                icon: Illustrations.PiggyBank,
+                translationKey: 'travel.features.saveMoney',
+                title: translate('travel.features.saveMoney'),
```
Comment (DylanDylann) [inline src/pages/Search/EmptySearchView.tsx:44]: Should we remove this title field and move tripsFeatures outside the component like in ManageTrips

**Comment by luacmartins** [comment]
@daledah conflicts

File: src/pages/Search/EmptySearchView.tsx (hunk 2 above)
Comment (DylanDylann) [reply src/pages/Search/EmptySearchView.tsx:93]: I think we should use FeatureList to DRY code, for your problem we can disable the button easily be remove some props

File: src/pages/Search/EmptySearchView.tsx [hunk 4]
```diff
@@ -1,43 +1,147 @@
...
+    const [travelSettings] = useOnyx(ONYXKEYS.NVP_TRAVEL_SETTINGS);
+    const [activePolicyID] = useOnyx(ONYXKEYS.NVP_ACTIVE_POLICY_ID);
+    const [primaryLogin = ''] = useOnyx(ONYXKEYS.ACCOUNT, {selector: (account) => account?.primaryLogin});
+    const policy = usePolicy(activePolicyID);
+    const [ctaErrorMessage, setCtaErrorMessage] = useState('');
+
+    const hasAcceptedTravelTerms = travelSettings?.hasAcceptedTerms;
+    const hasPolicyAddress = !isEmptyObject(policy?.address);
+
+    const onBookATripPress = useCallback(() => {
```
Comment (DylanDylann) [inline src/pages/Search/EmptySearchView.tsx:57]: This code comes from ManageTrips, right? If yes, let's create a Util function to DRY code

**Comment by DylanDylann** [comment]
@daledah Also please check this problem

**Comment by luacmartins** [comment]
@daledah what are the next steps here? Let's get this one to the finish line

**Comment by dubielzyk-expensify** [comment]
> @shawnborton The UI is updated, please test again.

Shawn's OOO so I'm testing :) Pulled it down locally and I see the same issue outlined by Shawn on the Travel empty state:

dev new expensify com_8082_search_q=type%3Aexpense%20status%3Aall%20sortBy%3Adate%20sortOrder%3Adesc
dev new expensify com_8082_search_q=type%3Aexpense%20status%3Aall%20sortBy%3Adate%20sortOrder%3Adesc (1)
dev new expensify com_8082_search_q=type%3Aexpense%20status%3Aall%20sortBy%3Adate%20sortOrder%3Adesc (2)

The travel page isn't filling up the space like the other two pages are.

**Comment by luacmartins** [comment]
Running another build [here](https://github.com/Expensify/App/actions/runs/11015500171)

**Review by dubielzyk-expensify — DISMISSED** [review]
Looks great. Thanks! Approving from a visual end, but I'll leave the code to the others 😄

**Comment by luacmartins** [comment]
@DylanDylann let's get this one reviewed

File: src/pages/Travel/ManageTrips.tsx [hunk 5]
```diff
@@ -67,20 +62,7 @@ function ManageTrips() {
                             setCtaErrorMessage(translate('travel.phoneError'));
```
Comment (DylanDylann) [inline src/pages/Travel/ManageTrips.tsx:62]: Also please put it into TripsResevationUtils.bookATrip

File: src/components/EmptyStateComponent/index.tsx [hunk 6]
```diff
@@ -82,7 +84,10 @@ function EmptyStateComponent({
     }, [headerMedia, headerMediaType, headerContentStyles, videoAspectRatio, styles.emptyStateVideo, lottieWebViewStyles]);
 
     return (
-        <ScrollView contentContainerStyle={[styles.emptyStateScrollView, {minHeight: minModalHeight}, containerStyles]}>
+        <ScrollView
+            contentContainerStyle={[!canEmptyViewBeScrolled && styles.emptyStateScrollView, {minHeight: minModalHeight}, !isSmallScreenHeight && styles.h100, containerStyles]}
```
Comment (DylanDylann) [inline src/components/EmptyStateComponent/index.tsx:88]: Let's use flexShrink

File: src/components/EmptyStateComponent/index.tsx [hunk 7]
```diff
@@ -82,7 +84,10 @@ function EmptyStateComponent({
...
+            style={[canEmptyViewBeScrolled && {flex: 1}]}
```
Comment (DylanDylann) [inline src/components/EmptyStateComponent/index.tsx:89]: Let's use styles.flex1

File: src/pages/Travel/ManageTrips.tsx [hunk 8]
```diff
@@ -63,24 +56,7 @@ function ManageTrips() {
...
-                            Navigation.navigate(ROUTES.TRAVEL_TCS);
-                            return;
-                        }
-                        if (ctaErrorMessage) {
-                            setCtaErrorMessage('');
-                        }
-                        Link.openTravelDotLink(activePolicyID)?.catch(() => {
-                            setCtaErrorMessage(translate('travel.errorMessage'));
-                        });
+                        TripsResevationUtils.bookATrip(translate, account?.primaryLogin ?? '', setCtaErrorMessage, ctaErrorMessage);
```
Comment (DylanDylann) [inline src/pages/Travel/ManageTrips.tsx:59]: Should we put account?.primaryLogin inside bookATrip function

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]

**Review by luacmartins — APPROVED** [review]

**Comment by sobitneupane** [comment]
The 'Copy URL to clipboard' context menu should appear on long press or double tap of the link. This functionality was not considered in this PR.

**Pull request merged**